tensorflow/compiler/xla/tests/exhaustive_op_test_utils.h

```diff
@@ -30,6 +30,26 @@ limitations under the License.
 namespace xla {
 using Eigen::half;
 
+namespace test_util {
+template <int N>
```

Comment:
After we remove the use of this template in the exhaustive_unary_test.cc, would it be better to put this back to the class as "private" or use anonymous namespace here?

Reply:
you still cannot have a specialization of a template class embedded inside a class in official c++.  so it has to remain in a separate namespace.

Reply:
although it could be an anonymous namespace as you say

Reply:
see comment above - i don't think that IntegralT can be removed from the FillInput function

Reply:
See my reply above - I think we should be able to remove IntegralT from FillInput.

Reply:
I understand that IntegralT is not int64, but we can still replace IntegralT with int64 here, because ConvertAndReplaceKnownIncorrectValueWith will ensure that we will retrieve the right bits from input_val. I am approving this and will make the change I am talking about later.